mvcc/index.go

```diff
@@ -91,10 +91,11 @@ func (ti *treeIndex) keyIndex(keyi *keyIndex) *keyIndex {
 func (ti *treeIndex) visit(key, end []byte, f func(ki *keyIndex)) {
 	keyi, endi := &keyIndex{key: key}, &keyIndex{key: end}
 
-	ti.RLock()
-	defer ti.RUnlock()
+	ti.Lock()
```

Comment:
@jingyih i think this improvement is still needed? can you make a separate PR for this change?

Reply:
Given that the Clone() is lazy clone, this should make read concurrency better. I agree we can add it.

On the other hand, in my benchmark I tried both with and without this change, and did not see noticeable difference. Maybe the read workload in my benchmark was not concurrent enough... Is there a typical number on how many concurrent read requests the backend is getting? Is it worth measuring?